gmock/build.gradle

```diff
@@ -1,45 +1,68 @@
-import org.apache.tools.ant.taskdefs.condition.Os
+import org.gradle.internal.os.OperatingSystem
+
 apply plugin: 'cpp'
 apply plugin: 'visual-studio'
 
 model {
-  toolChains {
-    if (Os.isFamily(Os.FAMILY_WINDOWS)) {
-      visualCpp(VisualCpp) {
-        def vs14Dir = file('C:\\Program Files (x86)\\Microsoft Visual Studio 14.0')
-        // If vs2015 is installed, fall back to vs2013 for now, until Gradle pulls in the right
-        // includes for the win10 sdk.
-        if (vs14Dir.exists()) {
-          def vs12Dir = file('C:\\Program Files (x86)\\Microsoft Visual Studio 12.0')
-          if (!vs12Dir.exists()) {
-            throw new GradleException('Compilation with VS 2015 is not currently supported by Gradle. Please install VS 2013, or use the CMake build option.')
-          }
-          installDir = vs12Dir
+    platforms {
+        x86 {
+            architecture 'x86'
+            operatingSystem OperatingSystem.current().getFamilyName()
+        }
+        x64 {
+            architecture 'x86_64'
+            operatingSystem OperatingSystem.current().getFamilyName()
         }
-      }
     }
-  }
-  components {
-    gmock(NativeLibrarySpec) {
-      sources {
-        cpp {
-          source {
-            srcDirs = ['src', 'gtest/src']
-            includes = ['*-all.cc']
-          }
-          exportedHeaders {
-            srcDirs = ['include', 'gtest/include', '.', 'gtest']
-            includes = ['**/*.h', '**/*.cc']
-          }
+    toolChains {
+        if (OperatingSystem.current().isWindows()) {
+            visualCpp(VisualCpp) {
+                // Workaround for VS2015 adapted from https://github.com/couchbase/couchbase-lite-java-native/issues/23
+                def VS_2015_INCLUDE_DIR = "C:/Program Files (x86)/Windows Kits/10/Include/10.0.10240.0/ucrt"
+                def VS_2015_LIB_DIR = "C:/Program Files (x86)/Windows Kits/10/Lib/10.0.10240.0/ucrt"
+
+                eachPlatform {
+                    cppCompiler.withArguments { args ->
+                        if (file(VS_2015_INCLUDE_DIR).exists()) {
```

Comment:
This basically tests if VS2015 is installed, not if it's being used to build.  On a system where both VS2013 and VS2015 are installed, will this break attempts to build with VS2013?  Is it possible to actually determine the compiler version from Gradle?